### ISSUE TITLE
Where is our algorithm DIVERSIFY in the diversify folder

### ISSUE THREAD

**Issue by DylTop** [microsoft/robustlearn#21]
Hello, thank you very much for disclosing your code, I have a question. My main concern is the OOD classification problem for time series, corresponding to the diversity folder. I didn't find any algorithms other than DANN, didn't I find them? Or will you add it later? Looking forward to your reply, thank you.

**Comment by jindongwang** [comment]
This folder is.
________________________________
发件人: DylTop ***@***.***>
发送时间: Sunday, April 9, 2023 9:02:48 PM
收件人: microsoft/robustlearn ***@***.***>
抄送: Subscribed ***@***.***>
主题: [microsoft/robustlearn] Where is our algorithm DIVERSIFY in the diversify folder (Issue #21)


Hello, thank you very much for disclosing your code, I have a question. My main concern is the OOD classification problem for time series, corresponding to the diversity folder. I didn't find any algorithms other than DANN, didn't I find them? Or will you add it later? Looking forward to your reply, thank you.

―

**Comment by DylTop** [comment]
Sorry, I don't quite understand what you mean. The image below is the result of running your code directly. The output I got was about training, validation, and testing of the DANN network. The Diversity algorithm you are referring to is in this folder, is it written incorrectly when this DANN outputs? I tried to modulate the code, but found that the algorithm used was TDBself. I don't quite understand a bit. Looking forward to your reply, thank you.

![1](https://user-images.githubusercontent.com/129808715/230805077-59c6e500-8d3f-4289-819c-9d9398cfa6ea.png)

**Comment by jindongwang** [comment]
Tdb is diversify, which is the codename during development.
________________________________
发件人: DylTop ***@***.***>
发送时间: Monday, April 10, 2023 8:49:37 AM
收件人: microsoft/robustlearn ***@***.***>
抄送: Jindong Wang ***@***.***>; Comment ***@***.***>
主题: Re: [microsoft/robustlearn] Where is our algorithm DIVERSIFY in the diversify folder (Issue #21)


Sorry, I don't quite understand what you mean. The image below is the result of running your code directly. The output I got was about training, validation, and testing of the DANN network. The Diversity algorithm you are referring to is in this folder, is it written incorrectly when this DANN outputs? I tried to modulate the code, but found that the algorithm used was TDBself. I don't quite understand a bit. Looking forward to your reply, thank you.

[1]<https://user-images.githubusercontent.com/129808715/230805077-59c6e500-8d3f-4289-819c-9d9398cfa6ea.png>

―

**Comment by DylTop** [comment]
Okay, I think I kind of got it. So if the diversity algorithm is used, that is, when the output of the print result, I use the red box circle is the wrong writing, right? But the accuracy of this output is so different from what you listed, what is the reason?

**Comment by jindongwang** [comment]
Target acc is the final acc. More details, @lw0517 can answer your questions.

**Comment by lw0517** [comment]
> Okay, I think I kind of got it. So if the diversity algorithm is used, that is, when the output of the print result, I use the red box circle is the wrong writing, right? But the accuracy of this output is so different from what you listed, what is the reason?

In fact, these three steps correspond to steps 2~4 in our paper. We choose the best one according to validation accuracy in step 4 during the whole training. Please note that these three steps are performed iteratively and repeatedly.

**Comment by DylTop** [comment]
Maybe my problem is not clearly described, can I explain it with Chinese? My English was poor enough to describe the problem I was trying to express.
Or can you find algorithms other than the Dann algorithm in the diversity folder? Didn't you compare the diversity algorithm with other algorithms on metrics such as accuracy? In particular, the diversity algorithm mentioned in the paper. The previous teacher replied that the diversity algorithm was in the diversity folder, but after the code was run, the result was not, and the corresponding algorithm in the folder was the DANN algorithm with poor performance.

**Comment by jindongwang** [comment]
@lw0517 I think we need to hear other people's opinions and make this folder clearer and better aligned with our paper. For instance, the method name, file name, output style, etc.

**Comment by lw0517** [comment]
> @lw0517 I think we need to hear other people's opinions and make this folder clearer and better aligned with our paper. For instance, the method name, file name, output style, etc.

Okay, I got it. I will revise the relevant descriptions in these two days.

**Comment by jindongwang** [comment]
@DylTop  Please wait for a few days. We are rerunning the program using latest Pytorch version. I know that you may not get the best results as readme using newer version of Pytorch. We have the training logs available online at: https://wandb.ai/luw12thu/diversify.

**Comment by jindongwang** [comment]
Hi @DylTop I have updated the code with support to new pytorch versions. Please update the repo and see the README file.

**Comment by DylTop** [comment]
Ok, thank you very much
